src/dmd/lib/package.d

```diff
@@ -34,21 +34,15 @@ private enum LOG = false;
 
 class Library
 {
-    static Library factory()
+    static Library factory(Target.ObjectFormat fmt)
     {
-        if (target.os == Target.OS.Windows)
+        final switch (fmt)
         {
-            return (target.mscoff || target.is64bit) ? LibMSCoff_factory() : LibOMF_factory();
+            case Target.ObjectFormat.COFF:  return LibMSCoff_factory();
+            case Target.ObjectFormat.ELF:   return LibElf_factory();
+            case Target.ObjectFormat.MachO: return LibMach_factory();
+            case Target.ObjectFormat.OMF:   return LibOMF_factory();
```

Comment:
Can we just ditch the functions and `return new ...` ?

Reply:
That would require the underlying classes to be `package` as opposed to `private`. There's no reason why not, but I'd prefer to do that separately in another PR